File: 백준/Silver/1912. 연속합/연속합.java
import java.util.*;
import java.io.*;

public class Main {
	public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        StringTokenizer st = new StringTokenizer(br.readLine());
        
        int max = -9999, num = -1000;
        for(int i = 0; i < n; i++){
            int tmp = Integer.parseInt(st.nextToken());
            
            num = Math.max(tmp, num + tmp);
            max = Math.max(max, num);
        }
        
        System.out.println(max);
	}
}

File: 백준/Silver/1912. 연속합/연속합.cc
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    int n; cin >> n;

    int sum = -2000, val = 0;
    for (int i = 0; i < n; i++) {
        int tmp; cin >> tmp;
        
        val = max(val + tmp, tmp);
        sum = max(val, sum);
    }
    
    cout << sum << endl;
    
    return 0;
}
